packages/core/ui-settings/core-ui-settings-common/src/ui_settings.ts

```diff
@@ -55,8 +55,12 @@ export interface UiSettingsParams<T = unknown> {
   optionLabels?: Record<string, string>;
   /** a flag indicating whether new value applying requires page reloading */
   requiresPageReload?: boolean;
-  /** a flag indicating that value cannot be changed */
-  readonly?: boolean;
+  /**
+   * a flag indicating that value cannot be changed
+   * 'strict' indicates that the value cannot be changed through API and is not displayed in the UI
+   * 'ui' indicates that the value is just not displayed in the UI
+   * */
+  readonly?: 'strict' | 'ui';
```

Comment:
Unless I'm mistaken, this isn't a passive change.  If so, I wouldn't change this, but rather leave it as a `boolean`and add an internal API-- `setReadOnlyMode`-- that changes how a read-only field is _set_.

cc: @lukeelmers

Reply:
I understand, I returned the `readonly` property to as it was and added a `readonlyMode` property instead. However, I'm not sure I understand correctly what `setReadOnlyMode` is supposed to do. Would this set the `readonlyMode` property to "strict"/"ui" when we're applying the allowlist, or is this supposed to be the setter that we would use specifically for updating `readonly` settings?